plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -206,29 +206,34 @@ private void updateModelGroup(
             source.put(MLModelGroup.DESCRIPTION_FIELD, updateModelGroupInput.getDescription());
         }
         if (StringUtils.isNotBlank(updateModelGroupInput.getName()) && !updateModelGroupInput.getName().equals(modelGroupName)) {
-            mlModelGroupManager.validateUniqueModelGroupName(updateModelGroupInput.getName(), ActionListener.wrap(modelGroups -> {
-                if (modelGroups != null
-                    && modelGroups.getHits().getTotalHits() != null
-                    && modelGroups.getHits().getTotalHits().value != 0) {
-                    for (SearchHit documentFields : modelGroups.getHits()) {
-                        String id = documentFields.getId();
-                        listener
-                            .onFailure(
-                                new IllegalArgumentException(
-                                    "The name you provided is already being used by another model with ID: "
-                                        + id
-                                        + ". Please provide a different name"
-                                )
-                            );
-                    }
-                } else {
-                    source.put(MLModelGroup.MODEL_GROUP_NAME_FIELD, updateModelGroupInput.getName());
-                    updateModelGroup(modelGroupId, source, listener);
-                }
-            }, e -> {
-                log.error("Failed to search model group index", e);
-                listener.onFailure(e);
-            }));
+            mlModelGroupManager
+                .validateUniqueModelGroupName(
+                    updateModelGroupInput.getName(),
+                    updateModelGroupInput.getTenantId(),
```

Comment:
Heh, I have this exact change on my local, looks like we're duplicating effort here.